core/src/main/java/io/grpc/Attributes.java

```diff
@@ -85,6 +96,17 @@ public Key(String name) {
     public String toString() {
       return name;
     }
+
+    /**
+     * Convinience method for creating instances of {@link Key} class.
+     *
+     * @param name of Key
+     * @param <T> Key type
+     * @return Key object
+     */
+    public static <T> Key<T> named(String name) {
```

Comment:
Is this really necessary, since the ctor is public?

```java
Key<Foo> myKey = new Key<>("bar");
Key<Foo> myKey = Key.named("bar");
```

Not much difference? If the latter is more aesthetically pleasing, maybe make the ctor private? (Not sure how folks feel about API-compatibility change there. But it is already marked experimental.)